pywr/notebook/__init__.py

```diff
@@ -17,8 +16,110 @@
 folder = os.path.dirname(__file__)
 with open(os.path.join(folder, "draw_graph.js"), "r") as f:
     draw_graph_template = Template(f.read())
+with open(os.path.join(folder, "save_graph.js"), "r") as f:
+    save_graph_template = Template(f.read())
 with open(os.path.join(folder, "graph.css"), "r") as f:
     draw_graph_css = f.read()
+with open(os.path.join(folder, "template.html"), "r") as f:
+    html_template = Template(f.read())
+
+
+class PywrSchematic:
+
+    def __init__(self, model, width=500, height=400, labels=False, attributes=False, css=None):
+        """This object contains methods that allow the graph of a pywr model network to be displayed in a jupyter
+        notebook or saved to an html file.
+
+        It also contains a method to save the node positions of a notebook graph back to a pywr model json file. Note
+        that this method currently odoes not work if the object has be instantiated using a model object. 
+
+        Parameters
+        ----------
+        model : pywr.core.Model or json-dict that describes a model
+            The model to display
+        width : int
+            The width of the svg canvas to draw the graph on
+        height : int
+            The height of the svg canvas to draw the graph on
+        labels : bool
+            If True, each graph node is labelled with its name. If false, the node names are displayed
+            during mouseover events
+        attributes : bool
+            If True, a table of node attributes is displayed during mouseover events
+        css : string
+            Stylesheet data to use instead of default
+        """
+
+        if isinstance(model, Model):
```

Comment:
You could instead have a classmethod for each of these functions. E.g. `from_json` and `from_model`. Or just make them static methods. I think most notebooks use `draw_graph` which has been broken here anyway?

Reply:
Do you mean converting the `to_d3` functions into static or class methods rather than standalone functions?